website/docs/r/stack_hci_network_interface.html.markdown

```diff
@@ -0,0 +1,128 @@
+
+# azurerm_stack_hci_network_interface
+
+Manages an Azure Stack HCI Network Interface.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-rg"
+  location = "West Europe"
+}
+
+resource "azurerm_stack_hci_logical_network" "example" {
+  name                = "example-hci-ln"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  custom_location_id  = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg1/providers/Microsoft.ExtendedLocation/customLocations/cl1"
+  virtual_switch_name = "ConvergedSwitch(managementcompute)"
+  dns_servers         = ["10.0.0.7", "10.0.0.8"]
+
+  subnet {
+    ip_allocation_method = "Static"
+    address_prefix       = "10.0.0.0/24"
+    route {
+      name                = "example-route"
+      address_prefix      = "0.0.0.0/0"
+      next_hop_ip_address = "10.0.20.1"
+    }
+    vlan_id = 123
+  }
+
+  tags = {
+    foo = "bar"
+  }
+}
+
+resource "azurerm_stack_hci_network_interface" "example" {
+  name                = "example-ni"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  custom_location_id  = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg1/providers/Microsoft.ExtendedLocation/customLocations/cl1"
+  dns_servers         = ["10.0.0.8"]
+
+  ip_configuration {
+    private_ip_address = "10.0.0.2"
+    subnet_id          = azurerm_stack_hci_logical_network.test.id
+  }
+
+  tags = {
+    foo = "bar"
+  }
+
+  lifecycle {
+    ignore_changes = [mac_address]
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Azure Stack HCI Network Interface. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Azure Stack HCI Network Interface should exist. Changing this forces a new resource to be created.
+
+* `location` - (Required) The Azure Region where the Azure Stack HCI Network Interface should exist. Changing this forces a new resource to be created.
+
+* `custom_location_id` - (Required) The ID of the Custom Location where the Azure Stack HCI Network Interface should exist. Changing this forces a new resource to be created.
+
+* `ip_configuration` - (Required) An `ip_configuration` block as defined below. Changing this forces a new resource to be created.
+
+* `dns_servers` - (Optional) A list of IPv4 addresses of DNS servers available to VMs deployed in the Network Interface. Changing this forces a new resource to be created.
+
+* `mac_address` - (Optional) The MAC address of the Network Interface. Changing this forces a new resource to be created.
+
+-> **Note:** If `mac_address` is not specified, it will be assigned by the server. If you experience a diff you may need to add this to `ignore_changes`.
```

Comment:
is there a reason we do not make this computed so the user doesn't have to ue ignore changes?

Reply:
From the [contribute doc](https://hashicorp.github.io/terraform-provider-azurerm/topics/best-practices/#setting-properties-to-optional-computed),
> In order to solve this, (new) fields should no longer be marked as Optional + Computed - instead where a split Create and Update method is used (see above) users can lean on ignore_changes to ignore values from a field with a default value, should they wish to continue using the default value.

 seems we are supposed to avoid optional+computed for new field.